include/onnxruntime/core/platform/ort_mutex.h

```diff
@@ -3,11 +3,103 @@
 
 #pragma once
 #ifdef _WIN32
+#include <Windows.h>
 #include <mutex>
-#include <condition_variable>
 namespace onnxruntime {
-using OrtMutex = std::mutex;
-using OrtCondVar = std::condition_variable;
+// Q: Why OrtMutex is better than std::mutex
+// A: OrtMutex supports static initialization but std::mutex doesn't. Static initialization helps us prevent the "static
+// initialization order problem".
+
+// Q: Why std::mutex can't make it?
+// A: VC runtime has to support Windows XP at ABI level. But we don't have such requirement.
+
+// Q: Is OrtMutex faster than std::mutex?
+// A: Sure
+
+class OrtMutex {
+ private:
+  SRWLOCK data_ = SRWLOCK_INIT;
+
+ public:
+  constexpr OrtMutex() = default;
+  // SRW locks do not need to be explicitly destroyed.
+  ~OrtMutex() = default;
+  OrtMutex(const OrtMutex&) = delete;
+  OrtMutex& operator=(const OrtMutex&) = delete;
+  void lock() { AcquireSRWLockExclusive(native_handle()); }
+  bool try_lock() noexcept { return TryAcquireSRWLockExclusive(native_handle()) == TRUE; }
+  void unlock() noexcept { ReleaseSRWLockExclusive(native_handle()); }
+  using native_handle_type = SRWLOCK*;
+
+  __forceinline native_handle_type native_handle() { return &data_; }
+};
+
+class OrtCondVar {
+  CONDITION_VARIABLE native_cv_object = CONDITION_VARIABLE_INIT;
+
+ public:
+  constexpr OrtCondVar() noexcept = default;
+  ~OrtCondVar() = default;
+
+  OrtCondVar(const OrtCondVar&) = delete;
+  OrtCondVar& operator=(const OrtCondVar&) = delete;
+
+  void notify_one() noexcept { WakeConditionVariable(&native_cv_object); }
+  void notify_all() noexcept { WakeAllConditionVariable(&native_cv_object); }
+
+  void wait(std::unique_lock<OrtMutex>& lk) {
+    if (SleepConditionVariableSRW(&native_cv_object, lk.mutex()->native_handle(), INFINITE, 0) != TRUE) {
+      std::terminate();
+    }
+  }
+  template <class _Predicate>
+  void wait(std::unique_lock<OrtMutex>& __lk, _Predicate __pred);
+
+  /**
+   * returns cv_status::timeout if the wait terminates when Rel_time has elapsed. Otherwise, the method returns
+   * cv_status::no_timeout.
+   * @param cond_mutex A unique_lock<OrtMutex> object.
+   * @param rel_time A chrono::duration object that specifies the amount of time before the thread wakes up.
+   * @return returns cv_status::timeout if the wait terminates when Rel_time has elapsed. Otherwise, the method returns
```

Comment:
nit: I find the name 'rel_time' to be a little unclear. would wait_duration or max_wait be better?